cohorts/load.py

```diff
@@ -533,8 +547,34 @@ def _load_single_patient_variants(self, patient, variant_type, merge_type, filte
             merged_variants = self._merge_variant_collections(dict(variant_collections), merge_type)
 
         self.save_to_cache(merged_variants, self.cache_names["variant"], patient.id, cached_file_name)
-
-        return filter_variants_with_metadata(merged_variants, filter_fn)
+        return on_return(merged_variants)
+
+    def _filter_variants_to_expressed(self, variants, patient, variant_type, merge_type):
+        """Given a VariantCollection, filters that collection to only expressed variants."""
+        if len(variants) == 0:
+            return variants
+
+        expressed_variants = []
+        expressed_metadata = {}
+        # TODO: we're currently using the same isovar cache that we use for expressed
+        # neoantigen prediction; so we pass in the same epitope lengths.
+        # This is hacky and should be addressed.
+        df_isovar = self._load_single_patient_isovar(patient=patient,
+                                                     variants=variants,
+                                                     epitope_lengths=[8, 9, 10, 11],
+                                                     variant_type=variant_type,
+                                                     merge_type=merge_type)
+        for i, row in df_isovar.iterrows():
+            genome = variants[0].ensembl
+            variant = Variant(contig=row["chr"],
+                              start=row["pos"],
+                              ref=row["ref"],
+                              alt=row["alt"],
+                              ensembl=genome)
+            if variant in variants:
```

Comment:
As opposed to caching the output? Yeah, I wanted to try to simplify by not caching this.